intro-to-functions/invoking-functions.md

```diff
@@ -0,0 +1,298 @@
+Traceback (most recent call last):
+  File "<stdin>", line 1, in <module>
+TypeError: len() takes exactly one argument (2 given)
+```
+
+You can recreate the error here:
+
+```python
+len("Hello", "World")
+```
+
+Reading through our error messages, we should determine:
+
+1. The type of error is `TypeError`
+1. The details state `len() takes exactly one argument (2 given)`. This function takes in one argument, but we gave it two.
+1. This error was traced to `line 1` from where this code was written.
+
+To debug, we should...
+
+1. re-run the code to make sure we can reproduce the error message
+1. read the error message and see where the erroneous code is
+1. find the line of code that calls this function
+1. adjust how we call the function, with the proper number of arguments passed in
+
+## Functions Return a Value Back to the Code that Invoked It
+
+Functions **_give back_** one value to the code that invoked the function. Often times, this value is the value that fulfills the function's responsibility.
+
+The value that a function **gives back** to whoever called it is called its **return value**.
+
+Return values can be any data type. They can even be `None`!
+
+We can usually describe a return value by simply describing what it represents. Return values always have a _literal_ value after invoking a function, though.
+
+Let's look at the return values for some functions we've observed, and some functions we can call.
+
+| Name of function | Description of Return Value | Line of code that invokes a function | Arguments passed in | The literal Return Value |
+| --- | --- | --- | --- | --- |
+`len` | length of the argument | `len(["Kelsea's Title", "Kelsea's Job Description])` | `["Kelsea's Title", "Kelsea's Job Description]` | 2
+`print` | nothing | `print("Garry's Models")` | `"Garry's Models"` | `None`
+`randint` | a random integer within the range of the two arguments | `random.randint(0, 777)` | `0` and `777` | 419
+`time` | the number of seconds since epoch | `time.time()` | - | `1602720584.8029282`
+
+Callout: `print`'s return value is always `None`! This is intentional. The people who defined `print` (aka Python) said that the responsibility of `print` is to return `None`.
+
+How do we figure out what a return value is or should be? The logic for determining the return value is inside of the function itself, so to figure out what the return value for any function is, we would need to...
+
+- look at how the function is defined, or
+- read the function's documentation
+
+### Calling Functions and Using Return Values is Our Job
+
+When a function is invoked, the return value is used to pass back a value to the code that invoked the function.
+
+Our logic will often rely on these return values of the functions we use.
+
+**Setting Up a Pattern:** We will often...
+1. _declare_ a variable
+1. then _assign_ that variable to the return value of a function, by invoking that function on the right-hand side of the assignment operator `=`.
```

Comment:
This list doesn't feel fixed to me-- It feels flexible that it might grow, shrink, or get refined!

Because of that, and because the lesson doesn't reference these items by number, I'd love to keep it at a generic "1. 1. 1." markdown list syntax. Lmk if you have thoughts on that!